evalml/pipelines/components/estimators/estimator.py

```diff
@@ -62,6 +64,8 @@ def predict(self, X):
         """
         try:
             X = infer_feature_types(X)
```

Comment:
For some reason, passing the equivalent RangeIndex to XGBoost's DMatrix results in the columns receiving an appended space when converted to strings.  e.g. ["0 ", "1 ", "2 "...]

Reply:
Bizzare.

Reply:
Ultimately this was an issue with xgboost and DMatrix, resolved in xgboost > 1.4.2.